rero_ils/modules/documents/utils_mef.py

```diff
@@ -0,0 +1,234 @@
+            click.echo('Found:')
+            for key in sorted(self.count_found.keys()):
+                click.echo(f'  {key:<20} : {self.count_found[key]}')
+        if self.count_exists:
+            click.echo('Exists:')
+            for key in sorted(self.count_exists.keys()):
+                click.echo(f'  {key:<20} : {self.count_exists[key]}')
+        if self.count_no_data:
+            click.echo('No Data:')
+            for key in sorted(self.count_no_data.keys()):
+                click.echo(f'  {key:<20} : {self.count_no_data[key]}')
+        if self.count_no_mef:
+            click.echo('No MEF:')
+            for key in sorted(self.count_no_mef.keys()):
+                click.echo(f'  {key:<20} : {self.count_no_mef[key]}')
+
+
+class ReplaceMefIdentifiedByContribution(ReplaceMefIdentifiedBy):
+    """Replace MEF identified by contribution class."""
+
+    def __init__(self, run=False, verbose=False):
+        """Constructor."""
+        self.mef_url = current_app.config.get('RERO_ILS_MEF_AGENTS_URL')
+        self.query = DocumentsSearch() \
+            .filter('exists', field='contribution.agent.identifiedBy')
+        self.cont_types = ['idref', 'gnd']
+        super().__init__(run=run, verbose=verbose)
+
+    def get_local(self, ref_type, ref_pid):
+        """Get local MEF record."""
+        return Contribution.get_contribution(ref_type, ref_pid)
+
+    def get_online(self, ref_type, ref_pid):
+        """Get online MEF record."""
+        ref = f'{ref_type}/{ref_pid}'
+        try:
+            # try to get the contribution online
+            data = Contribution._get_mef_data_by_type(ref_pid, ref_type)
+            metadata = data['metadata']
+            if metadata.get('idref') or metadata.get('gnd'):
+                self.increment_count(self.count_found, ref)
+                # delete mef $schema
+                metadata.pop('$schema', None)
+                # create local contribution
+                return Contribution.create(data=metadata, dbcommit=True,
+                                           reindex=True)
+            else:
+                # online contribution has no IdREf, GND or RERO
+                self.increment_count(self.count_no_data, ref)
+        except Exception as err:
+            # no online contribution found
+            self.increment_count(self.count_no_mef, ref)
+
+    def process(self):
+        """Process replacement."""
+        if self.verbose:
+            click.echo(f'Found: {self.query.count()}')
+        for hit in self.query.source('pid').scan():
+            doc = Document.get_record_by_id(hit.meta.id)
+            new_contributions = []
+            changed = False
+            for contribution in doc.get('contribution', []):
+                new_contributions.append(contribution)
+                ref_type = contribution['agent'].get(
+                    'identifiedBy', {}).get('type', '').lower()
+                if ref_type in self.cont_types + ['rero']:
```

Comment:
Why don't include `rero` when set up the `self.cont_types` array (into __init__ method) ?

Reply:
We will find RERO identifiedBy but we will replace it only with GND or IDREF $refs.